resource/cmake/CommonFunctions.cmake

```diff
@@ -0,0 +1,443 @@
+###############################################################################
```

Comment:
One more thought: I would suggest to avoid using `macro()`'s and to use `function()`'s whenever possible, since `macro()` can be trickier and, I think, don't have any real advantage over pure functions that I can think of.

Reply:
We could review the existing macros and turn them all into functions, unless there are good reasons to keep them as macros. The only one that I can think of is `rtigw_init_globals()`, and only because it sets a lot of variables, and even then, we should probably turn it into a function which sets all variables in the CACHE so that command line arguments don't have to be repeated over consecutive calls to `cmake <dir>`

Reply:
changed some macros to functions